### DISCUSSION TITLE
ScriptReference/EditorUtility.SetDirty

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#113] [Notes]
# ScriptReference/EditorUtility.SetDirty

https://docs.unity3d.com/ScriptReference/EditorUtility.SetDirty.html

**Comment by docscomments** [comment]
**Comment by Tom Gleadall:**  

If you aren't using a SerializedProperty yet you want to set a scene as dirty then you can use [`EditorSceneManager.MarkSceneDirty`](https://docs.unity3d.com/ScriptReference/SceneManagement.EditorSceneManager.MarkSceneDirty.html) or [`EditorSceneManager.MarkAllScenesDirty`](https://docs.unity3d.com/ScriptReference/SceneManagement.EditorSceneManager.MarkAllScenesDirty.html) instead.

**Comment by STARasGAMES** [comment]
You can track ALL SetDirty calls that happen to track down what might cause unwanted scene dirty. To do so go to "Help->About Unity" and while About Unity window is open type on your keyboard "internal". This enables a special editor mode. 
Then you go to "Preferences" window, "Diagnostics" tab, "Core" drop down, "ObjectSetDirty" set to true. This requires editor restart. 
Be warned that this decreases editor performance, so you want this option and internal mode enabled only for a time you debugging.

In the Console you will see logs with InstanceId of dirtied object inside [ ] brackets, also log should have stacktrace of the SetDirty call.
Also, there is EditorSceneManager.sceneDirtied event. Log something in the callback of this event. And when you see this log in the Console it means that previous "ObjectSetDirtyLog" causes scene to become dirty.

UPD 
disable "internal" mode the same way you enable it
